### DISCUSSION TITLE
Is there a way to delete temporary files?

### DISCUSSION THREAD

**Discussion by yaghmr** [onekey-sec/unblob#687] [Q&A] [answered]
Is there a way to delete _temporary files_? For example,` xz.uncompressed`, `rootfs.ext2`; each of these files has its respective directory with the extracted files.
The problem is that some chunks are whole (`chunk.is_whole_file` is true), but they should still be removed.

I am using version _23.10.31_ in the following manner: `unblob -f -v -d 10 -e output_unblob/ image.bin`

**Comment by qkaiser** [comment]
Can you provide a sample that we can use for reproduction ? Thanks :)

**Reply by yaghmr** [reply]
You can use this image: https://github.com/praetorian-inc/DVRF/blob/master/Firmware/DVRF_v03.bin

Here is an excerpt of the output:

```
4096            d ./32-7753760.trx_v1_extract
1648364         f ./32-7753760.trx_v1_extract/part0
4096            d ./32-7753760.trx_v1_extract/part0_extract
6105336         f ./32-7753760.trx_v1_extract/part1
4096            d ./32-7753760.trx_v1_extract/part1_extract
992             f ./7753760-7754752.unknown
32              f ./0-32.unknown
```

As you can see, even though `part0` and `part1` were correctly extracted, they were not deleted.

**Reply by qkaiser** [reply]
It's probably a regression from https://github.com/onekey-sec/unblob/issues/303, I'll have a look and open a detailed ticket about it. Thanks.

**Reply by qkaiser** [reply]
Work started at https://github.com/onekey-sec/unblob/pull/766

**Comment by qkaiser** [comment, marked as answer]
After some internal discussion, we agreed not to implement this auto-deletion. The rationale is that this deletion strategy could lead to issues after extraction, even more so with emulation in mind. A script or a symlink could reference an archive that got deleted since the extraction was successful, blocking the script from running as expected.

Also, we don't want to have this auto-deletion strategy enabled by default. This means we would need to add yet another command line switch that users can use to tell unblob they want to auto-delete successfully extracted files. This would definitely add more confusion to a [CLI that already needs some rethinking](https://github.com/onekey-sec/unblob/issues/243). Bringing even more confusion on the subject of files vs chunks, [carve vs extract](https://github.com/onekey-sec/unblob/issues/326).

Having `keep-extracted-chunks` as an opt-in to keep extracted chunks and, say, `auto-delete-files` as an opt-in to delete extracted files would bring confusion on many levels:
- using an option to keep thing and another one not to keep thing is counter-intuitive
- the difference between files and chunks (we don't expect users to naturally understand the difference)

We might revisit this decision in the future. In the meantime there's a patch you can apply at https://github.com/onekey-sec/unblob/pull/766/files

**Comment by yaghmr** [comment]
Binwalk has the option `-r, --rm    Delete carved files after extraction`, to do this. 
However, I acknowledge your decision and will continue to use Binwalk.
